src/org/zaproxy/zap/extension/spider/SpiderPanel.java

```diff
@@ -172,6 +178,25 @@ public SpiderPanel(ExtensionSpider extension, SpiderParam spiderScanParam) {
 		super("spider", new ImageIcon(SpiderPanel.class.getResource("/resource/icon/16/spider.png")), extension);
 
 		tabbedPane = new JTabbedPane();
+		tabbedPane.addChangeListener(new ChangeListener() {
+			@Override
+			public void stateChanged(ChangeEvent e) {
+				switch(tabbedPane.getSelectedIndex()) {
+				case 0:
+					getUrlsExportButton().setEnabled(true);

```

Comment:
This button should also be "re-enabled" when the Messages tab is hidden, otherwise, if you have the Messages tab selected and then hide it, the export button shown/enabled is the Messages one instead of URLs.